src/main/java/uk/ac/ebi/embl/api/validation/helper/ReferenceUtils.java

```diff
@@ -0,0 +1,152 @@
+            }
+            publication.addAuthors(authorList);
+        }
+        Reference ref = new ReferenceFactory().createReference();
+        ref.setPublication(publication);
+        ref.setAuthorExists(true);
+        ref.setLocationExists(true);
+        ref.setReferenceNumber(1);
+        return ref;
+    }
+    private  Publication getPublication(String address, Date date) {
+        Submission submission = (new ReferenceFactory()).createSubmission();
+        submission.setDay(date == null ? Calendar.getInstance().getTime(): date);
+        if(address != null) {
+            submission.setSubmitterAddress(address);
+        }
+        return submission;
+    }
+
+    private List<Person> getAuthors(String block) {
+        List<Person> authors = new ArrayList<>();
+        block = FlatFileUtils.remove(block, ';');
+        for (String author : FlatFileUtils.split(block, ",")) {
+            EmblPersonMatcher personMatcher = new EmblPersonMatcher(null);
+            if (!personMatcher.match(author)) {
+                authors.add(new ReferenceFactory().createPerson( author, null));
+            }
+            else {
+                authors.add(personMatcher.getPerson());
+            }
+        }
+        return authors;
+    }
+
+    boolean doAddAuthorsToConsortium(String submissionAccountId) {
+        return submissionAccountId != null && submissionAccountId.equalsIgnoreCase("Webin-55551");
+    }
+
+    public Reference constructReference(List<ReferenceEntity> referenceEntityList) throws UnsupportedEncodingException {
+        Publication publication = new Publication();
+        ReferenceFactory referenceFactory = new ReferenceFactory();
+        Reference reference = referenceFactory.createReference();
+        HashSet<String> consortium = new HashSet<>();
+        String pubConsortium = "";
+
+        for (ReferenceEntity refEntity : referenceEntityList) {
+            String pConsrtium = refEntity.getConsortium();
+            consortium.add(pConsrtium);
+
+            if (pConsrtium == null)//ignore first_name,middle_name and last_name ,if consortium is given : WAP-126
+            {
+                Person person = null;
+
+                person = referenceFactory.createPerson(
+                        EntryUtils.concat(" ", WordUtils.capitalizeFully(
+                                EntryUtils.convertNonAsciiStringtoAsciiString(refEntity.getSurname()), '-', ' '),
+                                EntryUtils.convertNonAsciiStringtoAsciiString(refEntity.getMiddleInitials())),
+                        getFirstName(EntryUtils.convertNonAsciiStringtoAsciiString(refEntity.getFirstName())));
+
+                publication.addAuthor(person);
+                reference.setAuthorExists(true);
+            }
+
+            Submission submission = referenceFactory.createSubmission(publication);
+            submission.setSubmitterAddress(getAddress(refEntity));
```

Comment:
Because we have merged submission contact and account info we are unnecessarily setting address more that once